src/app/[locale]/client-page.tsx

```diff
@@ -0,0 +1,39 @@
+'use client';
+
+import {useQuery} from '@tanstack/react-query';
+import {SearchResult} from '@/lib/dataset-fetcher';
+import DatasetCard from '@/components/dataset-card';
+import {Dataset} from '@/lib/dataset-fetcher';
+
+interface ClientPageProps {
+  initialDatasets: SearchResult;
+  locale: string;
+}
+
+export default function ClientPage({initialDatasets, locale}: ClientPageProps) {
+  const {data: {datasets} = {}}: {data: {datasets?: Array<Dataset>}} = useQuery(
+    {
+      queryKey: ['Datasets'],
+      queryFn: () =>
+        fetch('/api/fetch-datasets').then(response => response.json()),
```

Comment:
Would it be possible to use the `await` syntax here? `then()` is perfectly valid, but I prefer to use one convention.